client/options.go

```diff
@@ -194,8 +195,8 @@ func WithTLSClientConfigFromEnv() Opt {
 // (see [WithAPIVersionNegotiation]).
 func WithVersion(version string) Opt {
 	return func(c *Client) error {
```

Comment:
FWIW, I think we should start looking at some of these to return an error; I know the `DOCKER_API_VERSION` at the time was introduced as a "power-user" feature and by design didn't do any validation, but I think it's becoming commonly used, and became a much more public option; one that may warrant an actual error when invalid.

I left that for now to not change the behavior, but something we should consider.